Content.Shared/Damage/Components/DamageableComponent.cs

```diff
@@ -45,46 +46,56 @@ public class DamageableComponent : Component, IDamageableComponent, IRadiationAc
         public string DamageContainerId { get; set; } = DefaultDamageContainer;
 
         // TODO DAMAGE Cache this
-        [ViewVariables] public int TotalDamage => _damageList.Values.Sum();
-        [ViewVariables] public IReadOnlyDictionary<DamageGroupPrototype, int> DamageClasses => damageListToDamageGroup(_damageList);
-        [ViewVariables] public IReadOnlyDictionary<DamageTypePrototype, int> DamageTypes => _damageList;
+        [ViewVariables] public int TotalDamage => _damageDict.Values.Sum();
+        [ViewVariables] public IReadOnlyDictionary<DamageTypePrototype, int> DamagePerType => _damageDict;
+        [ViewVariables] public IReadOnlyDictionary<DamageGroupPrototype, int> DamagePerGroup => DamageGroupPrototype.DamageTypeDictToDamageGroupDict(_damageDict, ApplicableDamageGroups);
+        [ViewVariables] public IReadOnlyDictionary<DamageGroupPrototype, int> DamagePerSupportedGroup => DamageGroupPrototype.DamageTypeDictToDamageGroupDict(_damageDict, SupportedDamageGroups);
```

Comment:
why have two dicts for this? why would anything care about the 'damage in an unsupported group'

Reply:
I used bad naming conventiont here. DamagePerGroup does not include 'unsupported groups' , it should be "DamagePerApplicableGroup". The difference is just that DamagePerSupported group only contains damage groups that were explicitly supported by the damage container, not just any damage group which can damage it.
 
So far most code just cares about whether a group is applicable (i.e., that it has an effect), and not whether its 'explicitly supported'. DamagePerSupportedGroup is used less than DamagePerGroup. The only place it's currently used is in the health scanner window UI, which shows the players damage arranged by by explicitly supported groups, and then lumps other damage types into 'Miscelaneous' instead of showing all applicable groups (which might only contain one or two actually supported damage types).

Again, currently the yaml files are such that these two dicts should always be equal, because all aplicable groups are also explicitly supported. For now I'm gonna rename that dictionary, and maybe rename DamagePerSupportedGroup --> DamagePerExplicitlySupportedGroup. Though maybe that name is too long?